### PR TITLE
filter out repetitive actions

### PR DESCRIPTION
**Pull request by jhheider**
Pulled out repetitive jobs to reduce future breakage.

@mxcl think we should split these into their own repos to simplify the `uses:` lines (removing the need for explicit checkouts)?

### CODE REVIEW

**Comment by mxcl** [comment]
GHA checks out the repo the action is in (it doesn’t for reusable workflows, it does for actions), you don’t have to do it.

Though the path is github.action_path I believe.

Good work here.

**Comment by jhheider** [comment]
I believe it does for actions with their own repos, but passing by path, as we do here, requires a manual checkout.

File: .github/actions/get-platform/action.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    value: ${{ steps.platform.outputs.test-matrix }}
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v3
+      with:
+        repository: teaxyz/pantry.core
+        path: pantry.core
+    - uses: ./pantry.core/.github/actions/cache-brewkit
```
Comment (mxcl) [inline .github/actions/get-platform/action.yml:30]: so change it to: `- uses: teaxyz/pantry-core/actions/cache-brewkit@main`

though this name sucks, we don’t cache brewkit. maybe just call it `setup`

File: .github/actions/get-platform/action.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: Cache Brewkit
+description: sets up tea.xyz/brewkit with caching
```
Comment (mxcl) [inline .github/actions/get-platform/action.yml:2]: name is wrong

**Comment by mxcl** [comment]
Getting rid of `cache-prefix` since it seems like an anti pattern here. Unless I'm missing something?

**Comment by jhheider** [comment]
> Getting rid of `cache-prefix` since it seems like an anti pattern here. Unless I'm missing something?
> 
> edit: as in: the cache key is calculated using brewkit’s deno.json contents hash already, so separating it further seems inefficient. Unless I'm missing something.

Since different jobs might use different deps, the cache could be different. Certainly it's different between linux/mac due to different paths. So, the idea is that you have a cache for each job, since its usages shouldn't change (unless brewkit does).

I think after reflection, that the cache key should have been `${{ runner.os }}-deno-$job-{{hashFiles($path_to_brewkit, $path_to_scripts)}}`.

**Comment by jhheider** [comment]
Your additions to this decomposition look good. The main YML files had long-since grown unwieldy.

**Comment by jhheider** [comment]
Made some additions, cleaned up some issues. I feel we're getting close.

**Comment by jhheider** [comment]
There. Hopefully, this is pretty solid, and reusable for https://github.com/teaxyz/cli/issues/376

**Comment by jhheider** [comment]
Still not sure why `hashFiles` isn't returning correctly for get-platform... https://github.com/teaxyz/pantry.core/actions/runs/4218175706/jobs/7322493457

**Comment by jhheider** [comment]
That appears to fix the outstanding issues, except why the hashFiles() are returning empty.

**Comment by mxcl** [comment]
> Since different jobs might use different deps, the cache could be different. Certainly it's different between linux/mac due to different paths. So, the idea is that you have a cache for each job, since its usages shouldn't change (unless brewkit does).

Yeah but the hash of `deno.jsonc` should differentiate that right? I guess it's possible for a source file to request a rep explicitly, but generally we don’t do that.

Maybe you're right.

**Comment by mxcl** [comment]
Accidentally merged this rather than another PR I was working on 🤦🏻‍♂️ soz.

New one: https://github.com/teaxyz/pantry.core/pull/336 same branch

**Comment by mxcl** [comment]
ah you are right. Hmm, sucks. I wonder if github provide the `job-name` as a key.

**Comment by mxcl** [comment]
Like, do we even need to have a hash key? deno treats URLs as immutable. We just have to make sure we don’t use tags that move for our own stuff.

**Comment by jhheider** [comment]
As far as I know, yes. Because otherwise the cache will never invalidate.

**Comment by mxcl** [comment]
So like the cache on my computer? Which is never a problem for invalidating. With deno at least I mean.

**Comment by mxcl** [comment]
well do what you want. But as far as I can tell all we need is a cache per platform/os

**Comment by jhheider** [comment]
But each script will populate different items to the cache. And if we don't hash anything then today's cache items will never change (making it useless when we update deps), unless I'm sorely mistaken about how this cache action works.

See https://github.com/actions/cache/blob/main/examples.md#deno

**Comment by mxcl** [comment]
possibly I'm wrong that it updates the cache for a key if it is different from what it fetched. In which case ignore me.

**Comment by jhheider** [comment]
My understanding is that the cache never updates once written.

**Pull request merged**